.github/workflows/android-docs.yml

```diff
@@ -47,26 +47,20 @@ jobs:
 
       - name: Setup Gradle
         uses: gradle/actions/setup-gradle@v3
-      
-      - name: Switch to gh-pages branch
+
+      - name: Build and commit docs
         run: |
+          cd android
+          ./gradlew :source:dokkaHtml --no-configuration-cache --rerun-tasks
```

Comment:
Generate new docs on current branch before switching to gh-pages and copying them there.